src/passes/TraceCalls.cpp

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2024 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Instruments the build with code to intercept selected function calls.
+// This can be e.g. used to trace allocations (malloc, free, calloc, realloc)
+// and build tools for memory usage analysis.
+// The pass supports SIMD but the multi-value feature is not supported yet.
+//
+// Instrumenting void free(void*):
+
+// Instrumenting function `void* malloc(int32_t)` with a user-defined
+// name of the tracer `trace_alloc` and function `void free(void*)`
+// with the default name of the tracer `trace_free` (`trace_` prefix
+// is added by default):
+// wasm-opt --trace-calls=malloc:trace_alloc,free -o test-opt.wasm test.wasm
+//
+//  Before:
+//   (call $malloc
+//     (local.const 32))
+//   (call $free (i32.const 64))
+//
+//  After:
+//   (local $0 i32)
+//   (local $1 i32)
+//   (local $2 i32)
+//   (block (result i32)
+//     (call $trace_allocation
```

Comment:
```suggestion
//     (call $trace_alloc
```
To match the comment text, unless I've misunderstood?

Reply:
yep, that was my miss, thanks.